api/api.go

```diff
@@ -769,10 +769,17 @@ func (b *BlockChainAPI) prepareBlockResponse(
 	}
 	if len(transactions) > 0 {
 		totalGasUsed := hexutil.Uint64(0)
+		logs := make([]*types.Log, 0)
 		for _, tx := range transactions {
+			txReceipt, err := b.receipts.GetByTransactionID(tx.Hash)
+			if err != nil {
+				return nil, err
+			}
 			totalGasUsed += tx.Gas
+			logs = append(logs, txReceipt.Logs...)
 		}
 		blockResponse.GasUsed = totalGasUsed
+		blockResponse.LogsBloom = types.LogsBloom(logs)
```

Comment:
hmm could we just use receipt logs bloom that is already calculated?

Reply:
Yeah, we could do that. I was just being proactive, so in case we have multiple transactions per block, then the block `logsBloom` and the receipt's `logsBloom` will not be the same. A block might have more than one logs emitted.

Reply:
ohh you are right! let's keep your approach. If needed we will optimize by storing it in storage instead of calculating on demand, so maybe just add a todo, but avoid doing it for sake of simplicity.